### PR TITLE
Annotations: Fix alerting annotation coloring

### LINKED ISSUE #37304: NGAlert/LegacyAlert: State Annotations are not colored correctly
**What happened**:
When creating an alert in the new system that is linked to panel, the annotations for state changes do not have corresponding colors like they did in dashboard alerting, so it is hard to follow:

image

That image has "Pending", "Normal" and "Alerting" annotations. (Note: Technically they are ` Normal` with a prefixed space, but when I edit them the color stays the same, can look into fixing that when we know what the frontend is expecting.

**What you expected to happen**:

Them to be colored as before (need to check colors, I think red = alerting, yellow = pending, green = normal). 

**How to reproduce it (as minimally and precisely as possible)**:
Create a Grafana managed alert rule in the new NG system that is linked to dashboard via the Dashboard UID and Panel ID annotations on the rule (which are populated automatically if you create the rule via the dashboard UI).

**Anything else we need to know?**:
Not sure if this is frontend and/or backend yet. The are some field in the db like (prev_state, new_state, alert_id), but I have checked what the old system makes yet to see if shows up correctly when I edit the db. 

**Environment**:
- Grafana version: ddf5b65c51
- Data source type & version:
- OS Grafana is installed on:
- User OS & Browser:
- Grafana plugins:
- Others:

### PR DESCRIPTION
**Pull request by kylebrandt**
**What this PR does / why we need it**:
Adds more information to annotations, with the eventual goal of getting proper coloring for their states.

Before they were blue, now:

image

Also updated with ngalert states.

**Which issue(s) this PR fixes**:

Fixes https://github.com/grafana/grafana/issues/37304

TODO:
 - [x] Check that annotations from legacy alerting also look correct, although I expect they should.

**Special notes for your reviewer**:
`.nextState` doesn't make it into the frontend even when it is returned in the REST api call. I think something might have happened in the annotation migration to react, as the annotations don't show correctly when not using ngalerting as well. (Ryan fixed this part for me in the "more anno fields" merge commit.

### CODE REVIEW

**Comment by kylebrandt** [comment]
Possibly related to https://github.com/grafana/grafana/issues/37160 if someone is working on annotaitons.

**Comment by ryantxu** [comment]
FYI: with:
image
I think this PR ***fixes*** display of legacy alerts in 8.1!

File: public/app/plugins/panel/timeseries/plugins/AnnotationsPlugin.tsx [hunk 1]
```diff
@@ -66,7 +66,7 @@ export const AnnotationsPlugin: React.FC<AnnotationsPluginProps> = ({ annotation
           }
 
           let x0 = u.valToPos(annotation.time, 'x', true);
-          const color = theme.visualization.getColorByName(annotation.color);
+          var color = theme.visualization.getColorByName(annotation.color);
```
Comment (domasx2) [inline public/app/plugins/panel/timeseries/plugins/AnnotationsPlugin.tsx:69]: should probably remain `const`

File: public/app/features/annotations/standardAnnotationSupport.ts [hunk 2]
```diff
@@ -109,9 +109,17 @@ export const annotationEventNames: AnnotationFieldInfo[] = [
   {
     key: 'id',
   },
-  // { key: 'userId' },
-  // { key: 'login' },
-  // { key: 'email' },
+];
+
+// Properties we want passed though, but not exposed in the UI config builders
+const allAnnoEvents: AnnotationFieldInfo[] = [
```
Comment (domasx2) [inline public/app/features/annotations/standardAnnotationSupport.ts:115]: As far as I can see, it would make sense to add more items to `annotationEventNames` instead of creating `allAnnoEvents`. Also should probably add `prevState` and `data` to `AnnotationEvent` type instead of ignoring the type error with `as any`

Comment (kylebrandt) [reply public/app/features/annotations/standardAnnotationSupport.ts:115]: This is bit beyond me - @ryantxu know what needs to change perhaps?

(or if you know this is the right thing to do @domasx2  , feel free to push to this branch if you like)

Comment (ryantxu) [reply public/app/features/annotations/standardAnnotationSupport.ts:115]: @domasx2 -- The issue here is that 'annotations' and 'alerts' use the same infrastructure, but annotations do not support these fields and we do not want to expand the generic annotations API with them yet.

TBH, I was surprised that alerting even uses the same code path! and this was the least intrusive way I could find to just let the values pass though.

I'll add more comments about this



not really.  The `standardAnnotaitonSupport` is made for annotations --

**Review by ryantxu — APPROVED** [review]
:man_shrugging:

**Review by domasx2 — APPROVED** [review]
